### ISSUE TITLE
Add default error handler 

### ISSUE THREAD

**Issue by akm4** [gravitee-io/issues#2081]
It would be very helpful to have a policy or other mechanism catching unhandled errors during the message processing and executing user provided code. Something similair as "default" in Java. 

### Expected Behavior
On creating API in API Management user can provide a code that would be executed when any of API policies fail. For example this code can decide what type of error to return back depending on error type. Or to collect all the info from context and send as an alert to specific system. 

### Current Behavior
Every error should be handeld in the specific policy and requires policy rewriting that can be not the best option when different teams use the same policy but require different processing (like return HTTP codes)

### Possible Solution
The first thing coming to mind is the "handler" that catches all the exceptions (or probably even specific ones). That can be enriched with user's code. 

The other option is to add a paramater to policy where user can link an error handler policy or perhaps a service.  

### Context
Real world issues are: proccessing different upstream errors from different backends in uniform way, mapping the return codes and bodies to something suitable to show to request originator - without extra internal information. stacktraces, private info in upstream responses etc. Cna be called customized response/policy error filtering.

**Comment by brasseld** [comment]
It will be part of gravitee 1.25.
See Gateway responses:https://github.com/gravitee-io/issues/issues/972

**Comment by akm4** [comment]
Wow! looks like i'd better spend time on checking backlog more attentively then writing an issue ;) 

Anyway, the error mapper is probably just a part of default errror handler. Are there any ideas of how it will be implemented and how does it match with the whole thing i described?

**Comment by brasseld** [comment]
Ok, so as far as I understand, you're looking for a way to define policies on error.
Meaning that you may be able to define policies on request, response, but also on errors. Am I right ?
What would be the type of policy you would put to this error chain ? (now that we will provide response-template by error-code)

**Comment by akm4** [comment]
Right, the idea was to be able to define a chain of policies when something goes wrong. 

The functionality that is clearly seen now is: http code and body mapper/setter. The logic of mapper can be a bit sophisticated, like analyzing response body for getting the error code. 

The other requirement is sending events to custom monitoring system. Now we can do it by custom policy in request/response but not in error handler
